docs/book/en/src/dev/rust/wasinn.md

```diff
@@ -0,0 +1,158 @@
+# Neural Network for WASI
+
+In WasmEdge, we implemented the [WASI-NN](https://github.com/WebAssembly/wasi-nn) (Neural Network for WASI) proposal to allows WebAssembly programs to access machine learning(ML) functions, in the fashion of "graph" loader API. [WASI-NN](https://github.com/WebAssembly/wasi-nn) allows users to "load, bind, and evaluate" models by functions:
+
+* [`load`](https://github.com/WebAssembly/wasi-nn/blob/f72b983c4cc91ac575af6babc57b5bccb7db7ba9/phases/ephemeral/witx/wasi_ephemeral_nn.witx#L108-L118) a model using variable opaque byte arrays
+* [`init_execution_context`](https://github.com/WebAssembly/wasi-nn/blob/f72b983c4cc91ac575af6babc57b5bccb7db7ba9/phases/ephemeral/witx/wasi_ephemeral_nn.witx#L125-L129) and bind some tensors to it using [`set_input`](https://github.com/WebAssembly/wasi-nn/blob/f72b983c4cc91ac575af6babc57b5bccb7db7ba9/phases/ephemeral/witx/wasi_ephemeral_nn.witx#L134-L142)
+* [`compute`](https://github.com/WebAssembly/wasi-nn/blob/f72b983c4cc91ac575af6babc57b5bccb7db7ba9/phases/ephemeral/witx/wasi_ephemeral_nn.witx#L165) the ML inference using the bound context
+* retrieve the inference result tensors using [`get_output`](https://github.com/WebAssembly/wasi-nn/blob/f72b983c4cc91ac575af6babc57b5bccb7db7ba9/phases/ephemeral/witx/wasi_ephemeral_nn.witx#L147-L160)
+
+You can find more detail about the WASI-NN proposal in [Reference](#Reference).
+
+In this section, we will use [an Rust example project](https://github.com/gusye1234/Wasi-NN-Examples) ![badge](https://github.com/gusye1234/Wasi-NN-Examples/actions/workflows/main.yaml/badge.svg) to demonstrate how to use the WASI-NN api and run a image classification demo.
```

Comment:
Please PR to this repo:
https://github.com/second-state/WasmEdge-WASINN-examples
Thanks.